libraries/AP_InertialSensor/AP_InertialSensor_BMI160.cpp

```diff
@@ -407,15 +414,10 @@ void AP_InertialSensor_BMI160::_read_fifo()
     }
 }
 
-void AP_InertialSensor_BMI160::_poll_data()
+bool AP_InertialSensor_BMI160::_poll_data()
 {
-    if (!_dev->get_semaphore()->take_nonblocking()) {
```

Comment:
the lock is now implicit in the bus... and no other board uses this driver... we added it a while ago exactly for this board.